paddle/fluid/operators/where_op.cc

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/operators/where_op.h"
+
+namespace paddle {
+namespace operators {
+
+class WhereOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("Condition"),
+                   "Input(Condition) of WhereOp should not be null.");
+    PADDLE_ENFORCE(
+        ctx->GetInputDim("Condition").size() >= 1,
+        "Input(Condition) should have number of dimension at least 1");
+    PADDLE_ENFORCE(ctx->HasOutput("Out"),
+                   "Output(OUt) of WhereOp should not be null.");
+    ctx->SetOutputDim("Out", {-1, ctx->GetInputDim("Condition").size()});
+  }
+
+ protected:
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext& ctx) const override {
+    auto output_type = framework::proto::VarType::INT64;
+    return framework::OpKernelType(output_type, ctx.device_context());
+  }
+};
+
+class WhereOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("Condition", "A bool tensor whose rank is at least 1");
```

Comment:
Please use `Input`, but not `Condition` here. And please refer to [Operator's Parameter Name Convention](https://github.com/PaddlePaddle/FluidDoc/blob/release/1.2/doc/fluid/dev/name_convention.md)